### DISCUSSION TITLE
QUIC Protocol

### DISCUSSION THREAD

**Discussion by gcpmusic** [DNSCrypt/dnscrypt-proxy#1846] [Q&A]
Hi
I just learned about the Google QUIC Protocol and wander if DNSCrypt-Proxy is able to filter and block the connections.

Anyone ?

Thank you in advance.

**Comment by jedisct1** [comment]
Before using QUIC or any other protocol, a DNS query is made.

And dnscrypt-proxy handles DNS. So if a name is blocked, it will be blocked before the application decides to use HTTP, QUIC or something else.

**Reply by ignoramous** [reply]
How about DNSCrypt over QUIC?

Curious if QUIC's encryption layer would it make DNSCrypt more expensive? May be _Anonymized Relays_ could be used as a QUIC "tunnel" that proxies traffic to actual resolver. _Relays_ know the _Client_ but not the contents (since encrypted by the DNSCrypt protocol) and _Resolvers_ know the contents but not the _Client_.
